trieste/acquisition/function/active_learning.py

```diff
@@ -0,0 +1,245 @@
+            :exc:`ValueError` or :exc:`~tf.errors.InvalidArgumentError` if used with a batch size
+            greater than one.
+        """
+        return bichon_ranjan_criterion(model, self._threshold, self._alpha, self._delta)
+
+    def update_acquisition_function(
+        self,
+        function: AcquisitionFunction,
+        model: ProbabilisticModel,
+        dataset: Optional[Dataset] = None,
+    ) -> AcquisitionFunction:
+
+        return function  # no need to update anything
+
+
+def bichon_ranjan_criterion(
+    model: ProbabilisticModel,
+    threshold: float,
+    alpha: float,
+    delta: int,
+) -> TensorType:
+    r"""
+    Return the *bichon* criterion (:cite:`bichon2008efficient`) and *ranjan* criterion
+    (:cite:`ranjan2008sequential`) used in Expected feasibility aquisition function for active
+    learning of failure or feasibility regions.
+
+    The problem of identifying a failure or feasibility region of a function :math:`f` can be
+    formalized as estimating the excursion set, :math:`\Gamma* = \{ x \in X: f(x) \ge T\}`, or
+    estimating the contour line, :math:`C* = \{ x \in X: f(x) = T\}`, for some threshold :math:`T`
+    (see :cite:`bect2012sequential` for more details).
+
+    It turns out that probabilistic models can be used as classifiers for identifying where
+    excursion probability is larger than 1/2 and this idea is used to build many sequential
+    sampling strategies. We follow :cite:`bect2012sequential` and use a formulation which provides
+    a common expression for these two criteria:
+
+    .. math:: E[\max(0, (\alpha s(x))^\delta - |T - m(x)|^\delta)]$
+
+    Here :math:`m(x)` and :math:`s(x)` are the mean and standard deviation of the predictive
+    posterior of a probabilistic model. Bichon criterion is obtained when :math:`\delta = 1` while
+    ranjan criterion is obtained when :math:`\delta = 2`. :math:`\alpha>0` is another parameter
+    that acts as a percentage of standard deviation of the posterior around the current boundary
+    estimate where we want to sample. The goal is to sample a point with a mean close to the
+    threshold :math:`T` and a high variance, so that the positive difference in the equation above
+    is as large as possible.
+
+    Only batches of size 1 are allowed.
+
+    :param model: The probabilistic model of the objective function.
+    :param threshold: The failure or feasibility threshold.
+    :param alpha: The parameter which determines the neihbourhood around the estimated contour
+        line as a percentage of the posterior variance in which to allocate new points.
+    :param delta: The parameter identifying which criterion is returned, Bichon for value of 1
+        and Ranjan for value of 2.
+    """
+
+    @tf.function
+    def acquisition(x: TensorType) -> TensorType:
+
+        tf.debugging.assert_shapes(
+            [(x, [..., 1, None])],
+            message="This acquisition function only supports batch sizes of one.",
+        )
+
```

Comment:
do we need to assert again here if delta is 1 or 2?

Reply:
it seems superflous if builder is doing all the checks